### PR TITLE
Add ship tests and extra tests fixups

### PR DESCRIPTION
**Pull request by dvir001**
## About the PR
Added tests and tests fixups

## Breaking changes
Many ships PR's going to show real errors.

**Changelog**
N/A

### CODE REVIEW

File: .github/workflows/update-credits.yml [hunk 1]
```diff
@@ -4,51 +4,36 @@ on:
...
       - uses: actions/checkout@v3.6.0
         with:
           ref: master
-        
+
       - name: Get this week's Contributors
         shell: pwsh
-        env:
-          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
         run: Tools/dump_github_contributors.ps1 > Resources/Credits/GitHub.txt
```
Comment (whatston3) [inline .github/workflows/update-credits.yml:22]: This should probably be in its own file.

Comment (dvir001) [reply .github/workflows/update-credits.yml:22]: Do what you need man of stone

**Review by whatston3 — APPROVED** [review]
Looks alright to my eyes.

New credits page looks like this:
image

**Comment by dvir001** [comment]
Based

**Pull request merged**